pkg/reconciler/ingress/ingress.go

```diff
@@ -367,11 +378,25 @@ func (r *BaseIngressReconciler) reconcileIngress(ctx context.Context, ra Reconci
 	if err != nil {
 		return fmt.Errorf("failed to probe IngressAccessor %s/%s: %w", ia.GetNamespace(), ia.GetName(), err)
 	}
+
 	if ready {
 		lbs := getLBStatus(gatewayServiceURLFromContext(ctx, ia))
 		publicLbs := getLBStatus(publicGatewayServiceURLFromContext(ctx))
 		privateLbs := getLBStatus(privateGatewayServiceURLFromContext(ctx))
-		ia.GetStatus().MarkLoadBalancerReady(lbs, publicLbs, privateLbs)
+
+		desiredRoutes, err := oresources.MakeRoutes(ia, publicLbs)
+		if err != nil {
+			return fmt.Errorf("failed to generate routes: %w", err)
+		}
+		routes, err := r.reconcileRoutes(ctx, ia, desiredRoutes)
+		if err != nil {
+			return fmt.Errorf("failed to reconcile routes: %w", err)
+		}
+		if allRoutesReady(routes) {
+			ia.GetStatus().MarkLoadBalancerReady(lbs, publicLbs, privateLbs)
+		} else {
+			ia.GetStatus().MarkLoadBalancerPending()
```

Comment:
[`MarkLoadBalancerPending()`](https://github.com/openshift-knative/knative-serving-networking-openshift/blob/ee0c384b067ce27106933230335d27d7896fcf36/vendor/knative.dev/serving/pkg/apis/networking/v1alpha1/ingress_lifecycle.go#L76-L79) will print the reason as `Waiting for VirtualService to be ready` which is same with when prober failed so it would be mislead. We should use [`MarkIngressNotReady()`](https://github.com/openshift-knative/knative-serving-networking-openshift/blob/ee0c384b067ce27106933230335d27d7896fcf36/vendor/knative.dev/serving/pkg/apis/networking/v1alpha1/ingress_lifecycle.go#L82) instead?